src/main/scala/ch/zzeekk/spark/temporalquery/TemporalQueryUtil.scala

```diff
@@ -225,20 +263,21 @@ object TemporalQueryUtil extends LazyLogging {
   /**
    * left outer join
    */
-  private def temporalKeyLeftJoinImpl( df1:DataFrame, df2:DataFrame, keys:Seq[String], rnkExpressions:Seq[Column], additionalJoinFilterCondition:Column )
-                                     (implicit ss:SparkSession, hc:TemporalQueryConfig) = {
+  private def temporalKeyOuterJoinImpl( df1:DataFrame, df2:DataFrame, keys:Seq[String], rnkExpressions:Seq[Column], additionalJoinFilterCondition:Column, joinType:String = "full" )
```

Comment:
```suggestion
  private def temporalKeyOuterJoinImpl( df1:DataFrame, df2:DataFrame, keys:Seq[String], rnkExpressions:Seq[Column], additionalJoinFilterCondition:Column, joinType:String)
```
I suggest to not set a default value forjoinType. When using this function one should be forced to choose the join type.

Reply:
default value for joinType removed: https://github.com/zzeekk/spark-temporalquery/pull/16/commits/f7f3f48cc316eb731c30380803c2dd5979161fb1